server/test/lib/device/device.saveHistoricalState.test.js

```diff
@@ -0,0 +1,72 @@
+const EventEmitter = require('events');
+const { assert, expect } = require('chai');
+const Device = require('../../../lib/device');
+const StateManager = require('../../../lib/state');
+const { BadParameters } = require('../../../utils/coreErrors');
+
+const event = new EventEmitter();
+
+describe('Device saveHistoricalState', () => {
+  it('should save historical sate (new state)', async () => {
```

Comment:
add a test with a device which has no state

Reply:
I add the test with feature has not last_value_changed (and no last_value_string)